### PR TITLE
Abstract type areas

### PR DESCRIPTION
**Pull request by j-awn**
[CLEANLINESS] [STYLE]
## About the PR 

Some areas are now abstract types. As a rule, base areas (e.g. `/area/station`, `/area/station/medical`, `/area/station/crew_quarters/lounge`) are abstract types; where necessary, a `concrete` subtype of the base abstract area was added. 

A few areas were renamed for consistency (e.g. `/area/station/maintenance/NWMaint` -> `/area/station/maintenance/northwest`), and a few others were re-pathed as well to allow for their parent type to be abstract (e.g. `/area/station/maintenance/eastsolar` -> `/area/station/maintenance/solar/east`).

Where maps used the now-abstract base type as an area with a unique name (e.g. Wrestlemap's `/area/station/crew_quarters/lounge{name = "Luxury Seating"}`), a new sub-type was created. Wrestlemap and Fleet were the main offenders here.

`/area/station/chemistry` was moved to `/area/station/science/chemistry`; I'm sure a lot of spaghetti code that considered chemistry separately from the rest of research could now be reworked, but I haven't gone through and looked for that yet.

I checked that all of the maps which have overrides in `_std/__build.dm` compiled and ran without abstract_violation crashes; only four maps did not compile (donut2, trunkmap, chiron, and icarus), and none of those four maps compiled on the master branch. I did not check whether the unused maps I touched compiled, and I did check that all of the prefabs open in StrongDMM (is there a better way to check if they'll build?).

I also replaced the default space tile with `/area/concrete` (as opposed to `/area`), as any `/area/foo` being abstract makes the base `/area` abstract as well.

None of these changes should be visible to players.

## Why's this needed? 

Abstract area types will allow for searching filtered_concrete_typesof in the spythief bounty-list generation. There was no way to make one abstract area without touching every map file, so I went through all the area types and tried to make reasonable decisions about what should be abstract. This will also allow for some future optimization in a couple other places.

## Changelog

N/A

### CODE REVIEW

**Comment by Tarmunora** [comment]
haven't read details - I feel like `area/space` be more readable than `area/concrete`, etc

**Comment by j-awn** [comment]
> haven't read details - I feel like `area/space` be more readable than `area/concrete`, etc

Wherever it was necessary to make a concrete subtype of an abstract area without an obvious label (like `/area/station/communications/office` as the concrete type replacing the now-abstract `area/station/communications`), I just called it concrete. I kept that language for the direct concrete subtype of `/area`, but if you think the gain in readability would be worth breaking that consistency I'd be happy to change it

**Comment by ZeWaka** [comment]
I definitely think the space area should be `area/space`

File: code/area.dm [hunk 1]
```diff
@@ -1221,6 +1239,8 @@
 	sims_score = 100
 	sound_group = "vr"
 
+/area/sim/concrete
```
Comment (ZeWaka) [inline code/area.dm:1242]: why tho, useless

File: code/area.dm [hunk 2]
```diff
@@ -1365,34 +1411,43 @@ area/station/communications
...
+/area/station/maintenance/solar/north
 	name = "North Solar Maintenance"
 	icon_state = "SolarcontrolN"
-
+ABSTRACT_TYPE(/area/station/maintenance/inner)
 /area/station/maintenance/inner
 	name = "Inner Maintenance"
 	icon_state = "imaint"
 
+/area/station/maintenance/inner/concrete
```
Comment (ZeWaka) [inline code/area.dm:1439]: why tho, useless

File: code/area.dm [hunk 3]
```diff
@@ -1365,34 +1411,43 @@ area/station/communications
...
 // Donut 3 specific areas //
 
 // Civilian
-
+ABSTRACT_TYPE(/area/station/library)
 /area/station/library
 	name = "Library"
 	icon_state = "library"
 
+/area/station/library/main
```
Comment (ZeWaka) [inline code/area.dm:1449]: why tho, useless

File: code/area.dm [hunk 4]
```diff
@@ -1402,52 +1457,35 @@ area/station/communications
...
 	icon_state = "treatment_center"
 
 // Asylum
-
+ABSTRACT_TYPE(/area/station/medical/asylum)
 /area/station/medical/asylum
 	name = "Asylum Mini-Station"
 	icon_state = "blue"
 
+/area/station/medical/asylum/main
```
Comment (ZeWaka) [inline code/area.dm:1486]: why tho, useless

File: code/area.dm [hunk 5]
```diff
@@ -1573,12 +1620,18 @@ area/station/communications
...
 		name = "Boiler room"
 		icon_state = "orange"
-
+ABSTRACT_TYPE(/area/station/hallway)
 /area/station/hallway/
 	name = "Hallway"
 	icon_state = "hallC"
 	sound_environment = 10
 
+/area/station/hallway/concrete
```
Comment (ZeWaka) [inline code/area.dm:1629]: why tho, useless

File: code/area.dm [hunk 6]
```diff
@@ -1680,15 +1741,17 @@ area/station/hallway/starboardupperhallway
...
 		name = "Under Construction"
 
-
+ABSTRACT_TYPE(/area/station/mining)
 /area/station/mining
 	name = "Mining"
 	icon_state = "mining"
 	sound_environment = 10
 
+/area/station/mining/concrete
```
Comment (ZeWaka) [inline code/area.dm:1753]: why tho, useless

File: code/area.dm [hunk 7]
```diff
@@ -1706,7 +1773,12 @@ area/station/hallway/starboardupperhallway
 	sound_loop = 'sound/ambience/station/underwater/sub_bridge_ambi1.ogg'
 #endif
 
-area/station/seaturtlebridge
+/area/station/bridge/concrete
```
Comment (ZeWaka) [inline code/area.dm:1776]: why tho, useless

File: code/area.dm [hunk 8]
```diff
@@ -1805,11 +1903,20 @@ area/station/seaturtlebridge
 	icon_state = "yellow"
 	sound_environment = 0
 
+ABSTRACT_TYPE(/area/station/crew_quarters/radio)
 /area/station/crew_quarters/radio
 	name = "Radio Lab"
 	icon_state = "green"
 	sound_environment = 2
 
+/area/station/crew_quarters/radio/concrete
```
Comment (ZeWaka) [inline code/area.dm:1912]: why tho, useless

File: code/area.dm [hunk 9]
```diff
@@ -2434,11 +2567,14 @@ area/station/security/visitation
 	name = "Emergency Solar Array 3"
 	icon_state = "yellow"
 
+ABSTRACT_TYPE(/area/station/quartermaster)
 /area/station/quartermaster
 	name = "Quartermaster's"
 	icon_state = "quart"
 	workplace = 1
 
+/area/station/quartermaster/concrete
```
Comment (ZeWaka) [inline code/area.dm:2576]: why tho, useless

File: code/area.dm [hunk 10]
```diff
@@ -2470,38 +2606,49 @@ area/station/security/visitation
...
 
+ABSTRACT_TYPE(/area/station/janitor)
 /area/station/janitor
-	name = "Janitor's Office"
+	name = "Janitor's"
 	icon_state = "janitor"
 	sound_environment = 3
 	workplace = 1
 
+/area/station/janitor/office
```
Comment (ZeWaka) [inline code/area.dm:2616]: why tho

File: code/area.dm [hunk 11]
```diff
@@ -2470,38 +2606,49 @@ area/station/security/visitation
...
 
+ABSTRACT_TYPE(/area/station/science)
 /area/station/science
 	//name = "Research Outpost Zeta"
 	name = "Research Sector"
 	icon_state = "purple"
 	sound_environment = 3
 	workplace = 1
 
+/area/station/science/concrete
```
Comment (ZeWaka) [inline code/area.dm:2647]: why tho, useless

File: code/obj/item/clothing/gimmick.dm [hunk 12]
```diff
@@ -1158,6 +1159,8 @@
 						playsound(src.loc, "sound/items/coindrop.ogg", 50, 1, null, 2)
 						sleep(rand(2,5))
 
+/obj/item/clothing/gloves/ring/concrete
```
Comment (ZeWaka) [inline code/obj/item/clothing/gimmick.dm:1162]: whyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy

File: code/area.dm [hunk 13]
```diff
@@ -398,6 +399,8 @@
 		if(area_space_nopower(src))
 			power_equip = power_light = power_environ = 0
 
+/area/concrete // the base area you SHOULD be using for space/ocean/etc.
```
Comment (ZeWaka) [inline code/area.dm:402]: bad name, this isn't a sidewalk

**Review by ZeWaka — COMMENTED** [review]
honestly I got tired of copy pasting the same comment over and over

why do you make all these dumb useless `/concrete` subtypes.... if you want to have the umbrella area be instantiable, just don't make it abstract lol

File: code/obj/item/clothing/gimmick.dm (hunk 12 above)
Comment (Tarmunora) [reply code/obj/item/clothing/gimmick.dm:1162]: 💀

**Comment by ZeWaka** [comment]
could you update this PR?

**Comment by AffectedArc07** [comment]
This PR has caused major stability and availability issues for me (host of mapdiffbot).
image
This repository has been blacklisted until this PR has been dealt with. (Mapdiffbot has literally hard-locked the system 5 times now)

File: code/area.dm [hunk 14]
```diff
@@ -1697,6 +1789,10 @@ area/station/hallway/starboardupperhallway
 	name = "Mining Magnet Control Room"
 	icon_state = "miningp"
 
+/area/station/mining/cargo_staff_room
+	name = "Cargo Staff Room"
+
+ABSTRACT_TYPE(/area/station/bridge)
```
Comment (ZeWaka) [inline code/area.dm:1795]: why abstract bridge?

File: code/obj/item/clothing/gimmick.dm (hunk 12 above)
Comment (ZeWaka) [inline code/obj/item/clothing/gimmick.dm:1162]: still bad name, no need for this to be abstract, also ATOMIZE PRs

File: code/area.dm (hunk 14 above)
Comment (j-awn) [reply code/area.dm:1795]: it's a top-level station area; mining, medbay, bridge, etc. do you think it should be non-abstract? as a general rule I tried to keep `/area/station/foo` abstract

File: code/obj/item/clothing/gimmick.dm (hunk 12 above)
Comment (j-awn) [reply code/obj/item/clothing/gimmick.dm:1162]: name is bad, reason is not- `ABSTRACT_TYPE(/obj/item/clothing/gloves/ring/wizard)` has been established for months. anyway i fixed it by making the one instantiation of `ring` in the underwater ghost house prefab a `ring/gold`.

**Pull request merged**